### DISCUSSION TITLE
Switch language

### DISCUSSION THREAD

**Discussion by Vinicius-Madeira** [penguin-teal/gnome-openweather#48] [Q&A] [answered]
## Summary
After a few hours of unsuccessful attempts to change the language i'm here to ask for help.
I cloned the repo and tried to change a few things in the code to see if it'd work, but no progress so far.

### Question
Is there any solution to change the App's language (even if I have to hardcode the language that I'll be using)?

---
I'm not willing to change my system's language to anything other than english though.
Maybe I could try to add a language selector to the settings later, but for now a quick solution would be much appreciated.

**Comment by penguin-teal** [comment, marked as answer]
Sure! I've thought about making a language selector before, just haven't gotten around figuring out how (or if) to do it with GNU gettext.

The system language is measured using the `LANG` environment variable (plus some other locale variables). You can see your current locale with:
```shell
$ locale
```
(Note that you will see like 20 different variables print out, but changing the `LANG` variable will change them all)

To see all installed languages do:
```shell
$ locale -a
```
You'll see something like:
```
C
C.utf8
en_US.utf8
POSIX
zh_CN.utf8
```
`en_US.utf8` is UTF-8 American English and `zh_CN.utf8` is UTF-8 Chinese. If you don't see the language you want you will have to install it which is distro-specific (your locale must appear here for this method to work).

The `env` program allows you to run a command with modified environment variables. So all you need to do is run `./nest-test.sh` with a modified `LANG` variable. So this command here will run the test shell in the Chinese language. Switch `zh_CN.utf8` to the language of your choice. If you use a language you do not have downloaded it will fallback to the `C` locale.
```shell
$ env LANG=zh_CN.utf8 ./nest-test.sh
```
